cmd/expiration-mailer/main.go

```diff
@@ -188,11 +189,28 @@ func (m *mailer) sendNags(conn bmail.Conn, contacts []string, certs []*x509.Cert
 	return nil
 }
 
-func (m *mailer) updateCertStatus(ctx context.Context, serial string) error {
-	_, err := m.dbMap.WithContext(ctx).Exec(
-		"UPDATE certificateStatus SET lastExpirationNagSent = ? WHERE serial = ?",
-		m.clk.Now(), serial)
-	return err
+// updateLastNagTimestamps updates the lastExpirationNagSent column for every cert in
+// the given list. Even though it can encounter errors, it only logs them and
+// does not return them, because we always prefer to simply continue.
+func (m *mailer) updateLastNagTimestamps(ctx context.Context, certs []*x509.Certificate) {
+	qmarks := make([]string, len(certs))
+	params := make([]interface{}, len(certs)+1)
+	for i, cert := range certs {
+		qmarks[i] = "?"
+		params[i+1] = core.SerialToString(cert.SerialNumber)
+	}
+
+	query := fmt.Sprintf(
+		"UPDATE certificateStatus SET lastExpirationNagSent = ? WHERE serial IN (%s)",
+		strings.Join(qmarks, ","),
+	)
+	params[0] = m.clk.Now()
+
+	_, err := m.dbMap.WithContext(ctx).Exec(query, params...)
+	if err != nil {
+		m.log.AuditErrf("Error updating certificate status for %d certs: %s", len(certs), err)
+		m.stats.errorCount.With(prometheus.Labels{"type": "UpdateCertificateStatus"}).Inc()
```

Comment:
Question: should this instead be `.Add(len(certs))`? Previously it was equally counting "number of failed certs" and "number of failed sql operations"; now those are different.